From 00aa82c336f4dd762f5211286d4564aa89709a50 Mon Sep 17 00:00:00 2001
From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Mon, 21 May 2018 16:45:02 +0200
Subject: [PATCH] HACK: only build and install localedef

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>

Upstream: https://git.pengutronix.de/cgit/ptxdist/plain/patches/localedef-glibc-2.27/0001-HACK-only-build-and-install-localedef.patch?id=47116f66f411d4dadfce42c2fdd6d41b351ccfd4
Signed-off-by: Peter Seiderer <ps.report@gmx.net>
[bcressey:
  - "fixed" a build failure after 64924422a99690d147a166b4de3103f3bf3eaf6c
  - rebased to 2.40]
Signed-off-by: Ben Cressey <bcressey@amazon.com>
---
 Rules              | 14 ++++++++++----
 include/features.h |  2 +-
 locale/Makefile    |  5 -----
 3 files changed, 11 insertions(+), 10 deletions(-)

diff --git a/Rules b/Rules
index 9010c5d5b2..f5e5525061 100644
--- a/Rules
+++ b/Rules
@@ -223,10 +223,16 @@ binaries-shared-notests = $(filter-out $(binaries-pie) $(binaries-static), \
 				       $(binaries-all-notests))
 
 ifneq "$(strip $(binaries-shared-notests))" ""
-$(addprefix $(objpfx),$(binaries-shared-notests)): %: %.o \
-  $(sort $(filter $(common-objpfx)lib%,$(link-libc))) \
-  $(addprefix $(csu-objpfx),start.o) $(+preinit) $(+postinit)
-	$(+link)
+$(addprefix $(objpfx),$(binaries-shared-notests)): %: %.o
+	$(CC) -o $@ \
+	      $(sysdep-LDFLAGS) $(LDFLAGS) $(LDFLAGS-$(@F)) \
+	      $(combreloc-LDFLAGS) $(relro-LDFLAGS) $(hashstyle-LDFLAGS) \
+	      $(filter-out $(addprefix $(csu-objpfx),start.o \
+						     $(start-installed-name))\
+			   $(+preinit) \
+			   $(link-extra-libs) \
+			   $(common-objpfx)libc% $(+postinit),$^) \
+	      $(link-extra-libs)
 endif
 
 ifneq "$(strip $(binaries-shared-tests))" ""
diff --git a/include/features.h b/include/features.h
index 093de6f44c..3bf61cf56e 100644
--- a/include/features.h
+++ b/include/features.h
@@ -479,7 +479,7 @@
    to be disabled when building those functions, despite _GNU_SOURCE
    being defined.  */
 #if __GLIBC_USE (ISOC23)
-# define __GLIBC_USE_C23_STRTOL 1
+# define __GLIBC_USE_C23_STRTOL 0
 #else
 # define __GLIBC_USE_C23_STRTOL 0
 #endif
diff --git a/locale/Makefile b/locale/Makefile
index 2810f28605..85b69a0518 100644
--- a/locale/Makefile
+++ b/locale/Makefile
@@ -76,7 +76,6 @@ aux = \
   xlocale \
   # aux
 others = \
-  locale \
   localedef \
   # others
 #others-static = \
@@ -84,7 +83,6 @@ others = \
 # locale \
 # # others-static
 install-bin = \
-  locale \
   localedef \
   # install-bin
 extra-objs = \
@@ -96,9 +94,6 @@ extra-objs = \
 generated += C-translit.h
 before-compile += $(objpfx)C-translit.h
 
-extra-libs = libBrokenLocale
-extra-libs-others = $(extra-libs)
-
 libBrokenLocale-routines = broken_cur_max
 
 subdir-dirs = programs
-- 
2.47.0

